File: AdBoard.API/AdBoard.DAL/Entities/RefreshToken.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdBoard.DAL.Entities;

[Table("refresh_tokens")]
public class RefreshToken
{
    public long Id { get; set; }

    public DateTime ExpiryDate { get; set; }

    public string Token { get; set; } = null!;

    public bool IsRevoked { get; set; }

    public Guid UserId { get; set; }

    public User? User { get; set; }
}

File: AdBoard.Api.Spring/src/main/kotlin/ru/atom/adboard/dal/entities/RefreshToken.kt
package ru.atom.adboard.dal.entities

import jakarta.persistence.*
import lombok.Data
import java.util.*

@Entity
@Data
@Table(name = "refresh_tokens")
class RefreshToken (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,
    @Column(nullable = false, unique = true)
    var token: String,
    var expiryDate: Date,
    @OneToOne
    val user: User
)
{
    constructor(token: String, expiryDate: Date, user: User) : this(0, token, expiryDate, user)
}
